### ISSUE TITLE
clip option does not work as expected

### ISSUE THREAD

**Issue by haukemac** [ImageMagick/ImageMagick#5709]
### ImageMagick version

7.1.0-51

### Operating system

Linux

### Operating system, version and so on

centos 7

### Description

According to the docu the clip option should follow a clipping path:

```
-clip
Apply the clipping path if one is present.
If a clipping path is present, it is applied to subsequent operations.
```

Images can have different paths but only one can be of type clipping-path. Example image with more than one path, the last path is of type clipping path. With the clip option ImageMagick takes always the first available path and not the one with the correct type.

### Steps to Reproduce

`convert clippingpath.tif -alpha transparent -clip -alpha opaque  -trim  out.png`

should clip around the 6th path (rectangle) which is of type clipping path. 

### Images

[Uploading clippingpath.tif.zip…]()

**Comment by haukemac** [comment]
[clippingpath.tif.zip](https://github.com/ImageMagick/ImageMagick/files/9920230/clippingpath.tif.zip)

**Comment by snibgo** [comment]
> convert clippingpath.tif -alpha transparent -clip -alpha opaque -trim out.png
should clip around the 6th path (rectangle) which is of type clipping path.

I don't understand this. How do you know there are 6 paths? IM "`magick clippingpath.tif -verbose info:`" can see only one path.

Your command succeeds for me, with IM 7.1.0-49, making the background around the figure transparent.

As you use IM v7, I suggest using "magick", not "convert".

**Comment by haukemac** [comment]
You have to open the image in Adobe Photoshop, there you find 6 paths, but only the last one is of type clipping path. It is a bit difficult in Photoshop, but the one with the clipping path has a **bold** name

the correct has rectangle shaped clipping. I attach a screenshot.
<img width="805" alt="clippingpath" src="https://user-images.githubusercontent.com/47486748/199543039-9622ecb5-1077-4e95-9299-c8b2b7e32c0d.png">

**Comment by fmw42** [comment]
Imagemagick only uses the clipping path, not any of the other type paths.

**Comment by haukemac** [comment]
And exactly that is not working.

**Comment by snibgo** [comment]
> You have to open the image in Adobe Photoshop ...

Sorry, I don't have PS, so can't comment on that.

**Comment by fmw42** [comment]
Imagemagick is taking the first path in your list of paths.  I can confirm from looking at the paths in Photoshop. I do not think there is a current way to choose from the list of paths to specify which one to use.

Sorry, I do not know much about the difference in paths in Photoshop.

The IM developers will need to comment.

**Comment by haukemac** [comment]
The clipping path name is identified via exiftool:
`[Photoshop]     Clipping Path Name              : Online`

Also it works to address that path via the `-clip-path _id_` parameter.

**Comment by fmw42** [comment]
I believe that -clip just takes the first path in the list.

If you want to choose a path (for example #6), you can do:

```
magick identify -quiet -format "%[8BIM:1999,2998:#6]" clippingpath.tif |\
magick -quiet clippingpath.tif \( - -negate \) \
-alpha off -compose copy_opacity -composite \
-define tiff:alpha=associated \
clippingpath.png
```

![clippingpath](https://user-images.githubusercontent.com/13458218/199562834-8bfc3ee3-74a0-4a12-8d1d-1c08d2a98916.png)

**Comment by haukemac** [comment]
yes. but then I think that either the implementation or the documentation is wrong:
```
-clip
Apply the clipping path if one is present.
If a clipping path is present, it is applied to subsequent operations.
```

**Comment by axelrose** [comment]
To me it looks as if `-clip` is always taking the first found path, not considering if it has the extra Photoshop attribute or not.

**Comment by fmw42** [comment]
I believe that is what I said above. It was designed to just look at the first or only path. You can request an enhancement, but I showed above the way to do it otherwise, if you need another path in the list.

**Comment by axelrose** [comment]
Understood. The enhancement would be useful to have since otherwise I have an extra step to check for existing paths (in your case a `magick identify`, for me an `exiftool` call with some business logic).

The problem comes from different understanding what "clipping path" means. IM `-clip` takes every path but Photoshop and perhaps other image editors have "paths" as many as they like but only specific for clipping.

**Comment by fmw42** [comment]
This is not a bug. You will need to request an enhancement in the Discussions forum.  But the IM developers are very busy and I have no idea when they might get to it. You or anyone else can do the development and contribute it.

P.S. I know only a little about Photoshop. Is there only one formal "clip" path permitted or can there be any number of clip paths in the list of paths.  From the discussion above, not all the paths are clip paths. Can you clarify about the different paths and how many there can be?

**Comment by FredrikEngquist** [comment]
The -clip-path option was included many years ago (at my request) for exactly this situation.
-clip always uses the first path in any image.
Photoshop only stores information about which path is the actual clipping path in the 8BIM profile.

You are completely correct to use e.g. Exiftool to read the Clipping Path Name from the image and then using that name with -clip-path "path_name" as a direct substitute for -clip.

**Comment by fmw42** [comment]
Use -clip-path rather than -clip. And then use the path name.

`magick clippingpath.tif[0] -background none -fill none -transparent none -alpha transparent -clip-path "Online" -alpha opaque -strip clippingpath_clipped.jpg`

![clippingpath_clipped](https://user-images.githubusercontent.com/13458218/204030884-97e16364-599c-41e0-8aa1-e68ab516ec5d.jpg)


In the above, I have saved as JPG rather than tif, since this forum does not support uploading tif.

**Comment by haukemac** [comment]
Ok. thank you. We will use the `-clip-path` function.

Perhaps the docu should be amended.
